madeofhuman/maintenance-react
# #160900802 user and admin dashboard

#### What does this PR do?
- implement user and admin dashboard, allowing a user to view all their requests, and an admin to view all the requests in the system. Protect the user and admin routes against unauthorised access

#### Description of Task to be completed?
- implement user dashboard
- implement admin dashboard
- protect both routes against unauthorised access

#### How should this be manually tested?
- visit https://maintenance-react-staging.herokuapp.com to see the landing page
- click the sign up or log in buttons to sign up for a new account or log in
- a successful login will automatically redirect you to your dashboard, while an admin will be automatically redirected to the admin page
- accessing the admin route as a regular user will redirect you to your dashboard
- accessing either route without logging in will redirect you to the landing page to log in

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#160900667](https://www.pivotaltracker.com/story/show/160900667)
[#160900738](https://www.pivotaltracker.com/story/show/160900738)

#### Screenshots
<img width="1458" alt="screen shot 2018-10-04 at 1 26 29 am" src="https://user-images.githubusercontent.com/7345686/46446502-984b3700-c774-11e8-9393-f371070dd20c.png">


#### Questions:
N/A

[Delivers #160900802 #160901759]
